src/TConsole.cpp

```diff
@@ -284,10 +285,12 @@ TConsole::TConsole(Host* pH, bool isDebugConsole, QWidget* parent)
     baseHFrameLayout->setMargin(0);
     centralLayout->setMargin(0);
 
-    mpCommandLine = new TCommandLine(pH, this, mpMainDisplay);
-    mpCommandLine->setContentsMargins(0, 0, 0, 0);
-    mpCommandLine->setSizePolicy(sizePolicy);
-    mpCommandLine->setFocusPolicy(Qt::StrongFocus);
+    if (!mIsDebugConsole && !mIsSubConsole) {
```

Comment:
I am guessing but I suppose these tests are sufficient to identify the "main" console - there are instances of `TConsole` also used for the *Errors* window in the Editor - and, IIRC, also for each invisible Buffer to hold off-screen text...

💡  I think that someone ought to add a classification `enum` to this class to identify which one of these each instance is for - it would be cleaner that the current incomplete set of `bool` flags also including, besides these two, `TConsole::mUserConsole` - there is a further `TConsole::mWindowIsHidden` but that is declared, initialized to `false` but never used (I guess it might have been inteded for buffer instances):
+ Central debug console (one for whole application)
+ main profile console (one per profile)
+ sub-consoles overlaid on main profile console (zero to many per profile, each with a unique name - maybe unique across "User Windows" and labels and buffers - but maybe not)
+ floating/dockable "User Window" (zero to many per profile, each with a unique name - maybe unique across "sub-consoles and labels and buffers - but maybe not)
+ buffers (zero to many per profile, each with a unique name - maybe unique across "sub-consoles and "User-windows" and labels - but maybe not)
+ Editor's Error console (one per profile)

Reply:
That sounds like a good idea.